lib/code_corps/github/sync/sync.ex

```diff
@@ -99,22 +103,94 @@ defmodule CodeCorps.GitHub.Sync do
+      count -> map |> Map.put(key, count)
+    end
+  end
+
+  @spec sync_repo(GithubRepo.t) :: {:ok, GithubRepo.t}
+  def sync_repo(%GithubRepo{} = repo) do
+    with {:ok, repo} <- repo |> mark_repo("fetching_pull_requests"),
+         {:ok, pr_payloads} <- repo |> GitHub.API.Repository.pulls |> sync_step(:fetch_pull_requests),
+         {:ok, repo} <- repo |> mark_repo("syncing_github_pull_requests", [syncing_pull_requests_count: pr_payloads |> Enum.count]),
+         {:ok, _pull_requests} <- pr_payloads |> Enum.map(&Sync.PullRequest.GithubPullRequest.create_or_update_pull_request(repo, &1)) |> ResultAggregator.aggregate |> sync_step(:sync_pull_requests),
+         {:ok, repo} <- repo |> mark_repo("fetching_issues"),
+         {:ok, issue_payloads} <- repo |> GitHub.API.Repository.issues |> sync_step(:fetch_issues),
+         {:ok, repo} <- repo |> mark_repo("syncing_github_issues", [syncing_issues_count: issue_payloads |> Enum.count]),
+         {:ok, _issues} <- issue_payloads |> Enum.map(&Sync.Issue.GithubIssue.create_or_update_issue(repo, &1)) |> ResultAggregator.aggregate |> sync_step(:sync_issues),
+         {:ok, repo} <- repo |> mark_repo("fetching_comments"),
+         {:ok, comment_payloads} <- repo |> GitHub.API.Repository.issue_comments |> sync_step(:fetch_comments),
+         {:ok, repo} <- repo |> mark_repo("syncing_github_comments", [syncing_comments_count: comment_payloads |> Enum.count]),
+         {:ok, _comments} <- comment_payloads |> Enum.map(&Sync.Comment.GithubComment.create_or_update_comment(repo, &1)) |> ResultAggregator.aggregate |> sync_step(:sync_comments),
+         {:ok, repo} <- repo |> mark_repo("receiving_webhooks")
+    do
+      {:ok, repo}
+    else
+      {:error, :fetch_pull_requests} -> repo |> mark_repo("errored_fetching_pull_requests")
+      {:error, :sync_pull_requests} -> repo |> mark_repo("errored_syncing_pull_requests")
+      {:error, :fetch_issues} -> repo |> mark_repo("errored_fetching_issues")
+      {:error, :sync_issues} -> repo |> mark_repo("errored_syncing_issues")
+      {:error, :fetch_comments} -> repo |> mark_repo("errored_fetching_comments")
+      {:error, :sync_comments} -> repo |> mark_repo("errored_syncing_comments")
+    end
+  end
+
+  def sync_project_github_repo(%ProjectGithubRepo{} = project_github_repo) do
+    %ProjectGithubRepo{github_repo: %GithubRepo{} = repo} = project_github_repo =
+      project_github_repo
+      |> preload_project_github_repo
+
+    with {:ok, project_github_repo} <- project_github_repo |> mark_project_repo("syncing_github_repo"),
+         {:ok, %GithubRepo{sync_state: "receiving_webhooks"}} <- repo |> sync_repo(),
+         project_github_repo <- Repo.get(ProjectGithubRepo, project_github_repo.id) |> preload_project_github_repo(),
+         {:ok, project_github_repo} <- project_github_repo |> mark_project_repo("syncing_tasks"),
+         {:ok, _tasks} <- project_github_repo |> Sync.Issue.Task.sync_project_github_repo() |> sync_step(:sync_tasks),
```

Comment:
Trying to read `Sync.Issue.Task.sync_project_github_repo` makes me think we may have nested too deep.

```Elixir
project_github_repo |> Sync.TaskSyncer.sync
```

Looks a bit clearer, but I'm still not happy with it. However, the we could alias `Sync.TaskSyncer` and just call 

```Elixir
project_github_repo |> TaskSyncer.sync
```

What do you think about flattening out our Sync namespace that way? Nesting `Task` and `GithubIssue` under `Issue` together seems like we did just for the sake of them being semantically related, but logically, we probably want the two processes as decoupled as possible.

```
sync/sync.ex
sync/task_syncer.ex
sync/github_issue_syncer.ex
```

etc. looks nicer to me.

For now, as part of this PR, I'm thinking we 

```Elixir
alias CodeCorps.GitHub.Sync.Issue.Task, as: TaskSyncer
```

and do the same for other modules. What do you think?